### PR TITLE
Prevent multiple WSAStartup() call

### PR DESCRIPTION
**Pull request by vkedwardli**
Still have no idea how can this be related to https://github.com/inada-s/gdxsv/issues/132
But let's try?

### CODE REVIEW

**Comment by inada-s** [comment]
I didn't know that there are side effects calling WSAStartup multiple times while the process is running.

**Comment by inada-s** [comment]
I want to call WSAStartup only once from the start of the process.

**Comment by vkedwardli** [comment]
May be let's wait for the test from オージー first?
I cannot reproduce it directly, full details: https://github.com/flyinghead/flycast/issues/281

**Comment by inada-s** [comment]
Thank you for the survey report.


I remembered a cmake warning log in actions build.
```
[ 99%] Building CXX object CMakeFiles/flycast.dir/core/windows/winmain.cpp.obj
In file included from D:\a\flycast\flycast\core\windows\winmain.cpp:22:
C:/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/x86_64-w64-mingw32/include/winsock2.h:15:2: warning: #warning Please include winsock2.h before windows.h [-Wcpp]
 #warning Please include winsock2.h before windows.h
  ^~~~~~~
[100%] Building CXX object CMakeFiles/flycast.dir/core/windows/win_vmem.cpp.obj
```

**Comment by inada-s** [comment]
This warning is not appear in upstream build.
Probably due to the addition of includes for a platform-specific fix (http request for version update check?).

**Comment by vkedwardli** [comment]
It is for the connection medium checking, since the interface info is available in ws2ipdef.h (which depends on winsock2)

**Comment by inada-s** [comment]
No the warning found, looks good.

File: core/windows/winmain.cpp [hunk 1]
```diff
@@ -431,6 +431,11 @@ void os_CreateWindow()
 	CreateMainWindow();
 	InitRenderApi();
 #endif	// !USE_SDL
+	WSADATA wsaData;
+	if (WSAStartup(MAKEWORD(2, 0), &wsaData) != 0) {
+		ERROR_LOG(COMMON, "WSAStartup failed. errno=%d", WSAGetLastError());
+		return;
+	}
```
Comment (inada-s) [inline core/windows/winmain.cpp:438]: Nice!

**Pull request merged**